JMS/com/ibm/mq/samples/jms/JmsResponse.java

```diff
@@ -131,6 +132,7 @@ private static void replyToMessage(JMSContext context, Message receivedMessage,
                 }
             }
 
+            context.rollback();

```

Comment:
Line 131 (return) means that the rollback won't happen for certain error cases e.g., MQRC_UNKNOWN_OBJECT_NAME